### PR TITLE
add org.kde.krename

### PR DESCRIPTION
**Pull request by Amit0617**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Amit0617** [comment]
bot, build org.kde.krename

**Comment by Amit0617** [comment]
bot, build org.kde.krename

File: org.kde.krename.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+    "id": "org.kde.krename",
+    "branch": "master",
```
Comment (hfiguiere) [inline org.kde.krename.json:3]:
```suggestion
```

**Comment by Amit0617** [comment]
bot, build org.kde.krename

**Comment by Amit0617** [comment]
```
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• tag-missing           : <release> required
Validation of files failed
```
i have added a `appdata.patch` but it is not working.

File: appdata.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
+diff --git a/src/org.kde.krename.appdata.xml b/src/org.kde.krename.appdata.xml
+index 49f4d31..f6e7f34 100644
+--- a/src/org.kde.krename.appdata.xml
++++ b/src/org.kde.krename.appdata.xml
+@@ -112,5 +112,10 @@
+   <project_group>KDE</project_group>
+   <provides>
+     <binary>krename</binary>
++    <launchable type="desktop-id">org.kde.krename.desktop</launchable>
++    <content_rating type="oars-1.1"/>
```
Comment (hfiguiere) [inline appdata.patch:10]: this is in the wrong place in the file. Definitely not inside `<provide>`. That' why the build still fails.

Comment (Amit0617) [reply appdata.patch:10]: so i should close provide after binary tag?

**Comment by Amit0617** [comment]
bot, build org.kde.krename

**Comment by Amit0617** [comment]
bot, stop

**Comment by hfiguiere** [comment]
the patch doesn't apply

**Comment by nedrichards** [comment]
@flathub/kde FYI

File: org.kde.krename.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            ]
+        },
+        {
+            "name": "kjs",
+            "config-opts": [ "-DENABLE_TESTING=OFF" ],
+            "buildsystem": "cmake-ninja",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://invent.kde.org/frameworks/kjs.git/",
```
Comment (tsdgeos) [inline org.kde.krename.json:41]: tarball not git

File: org.kde.krename.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                    "tag": "v5.85.0"
+                }
+            ]
+        },
+        {
+            "name": "krename",
+            "buildsystem": "cmake-ninja",
+            "sources": [
+                { "type": "git",
+                  "commit" : "c5a07a250095c65edb89f90e131a1f1fca3b3c7b",
```
Comment (tsdgeos) [inline org.kde.krename.json:51]: please use tarball not git

File: org.kde.krename.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
+{
+    "id": "org.kde.krename",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15-21.08",
+    "sdk": "org.kde.Sdk",
+    "command": "krename",
+    "rename-icon": "krename",
+    "finish-args": ["--share=ipc", "--socket=fallback-x11", "--socket=wayland", "--filesystem=host" ],
```
Comment (travier) [inline org.kde.krename.json:8]: Those should be on their own lines

Comment (Amit0617) [reply org.kde.krename.json:8]: what do you mean by "those"?

**Comment by Amit0617** [comment]
bot, build org.kde.krename

File: org.kde.krename.json (hunk 5 above)
Comment (travier) [reply org.kde.krename.json:8]: Each element from the array in finish-args should be on its own line.

File: org.kde.krename.json [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            ]
+        },
+        {
+            "name": "kjs",
+            "config-opts": [ "-DENABLE_TESTING=OFF" ],
+            "buildsystem": "cmake-ninja",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.kde.org/Attic/frameworks/5.83/portingAids/kjs-5.83.0.tar.xz",
```
Comment (travier) [inline org.kde.krename.json:41]: The latest version is at https://download.kde.org/stable/frameworks/5.89/portingAids/

File: org.kde.krename.json [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+                {
+                    "type": "archive",
+                    "url": "https://taglib.github.io/releases/taglib-1.12.tar.gz",
+                    "sha256": "7fccd07669a523b07a15bd24c8da1bbb92206cb19e9366c3692af3d79253b703"
+                }
+            ]
+        },
+        {
+            "name": "kjs",
+            "config-opts": [ "-DENABLE_TESTING=OFF" ],
```
Comment (travier) [inline org.kde.krename.json:41]: All array members should be on their own line. Same for above in taglib.

Comment (Amit0617) [reply org.kde.krename.json:41]: yeah they are in there separate line already. If i am understanding correctly that you are asking for different elements in sources array.

Comment (travier) [reply org.kde.krename.json:41]: Yes

**Comment by travier** [comment]
```
FB: Running: patch -p1 -i /srv/buildbot/worker/build-x86_64-5/build/appdata.patch
patching file src/org.kde.krename.appdata.xml
Hunk #1 FAILED at 112.
1 out of 1 hunk FAILED -- saving rejects to file src/org.kde.krename.appdata.xml.rej
```
There is something wrong with your patch here.

**Comment by Amit0617** [comment]
bot, build org.kde.krename

**Comment by Amit0617** [comment]
```zgrep "<icon type=\"remote\">" builddir/*/share/app-info/xmls/org.kde.krename.xml.gz || test -f builddir/*/share/app-info/icons/flatpak/128x128/org.kde.krename.png
 in dir /srv/buildbot/worker/build-x86_64-7/build (timeout 1200 secs)
 watching logfiles {}
 argv: b'zgrep "<icon type=\\"remote\\">" builddir/*/share/app-info/xmls/org.kde.krename.xml.gz || test -f builddir/*/share/app-info/icons/flatpak/128x128/org.kde.krename.png'
 using PTY: False
program finished with exit code 1
elapsedTime=0.014501```

**Comment by Amit0617** [comment]
i checked this file location `builddir/*/share/app-info/xmls/org.kde.krename.xml.gz` in that file there is no `<icon type=\"remote\">` but intead `<icon type="cached" height="64" width="64"` is there. 
Checked this location `builddir/*/share/app-info/icons/flatpak/128x128/org.kde.krename.png` too. but it is not available there but `builddir/*/share/app-info/icons/flatpak/64x64/org.kde.krename.png` is there.

**Comment by hfiguiere** [comment]
an 128x128 icon is required. make sure there is one or a svg, installed in the standard locations.

**Comment by hfiguiere** [comment]
looking here there is only 16, 32 and 48 pixels

https://invent.kde.org/utilities/krename/-/tree/master/src/pics

**Comment by Amit0617** [comment]
> looking here there is only 16, 32 and 48 pixels
> 
> https://invent.kde.org/utilities/krename/-/tree/master/src/pics

it's strange from where then 64x64 pixel image is coming

**Comment by travier** [comment]
You can find the SVG icons in https://invent.kde.org/frameworks/breeze-icons and generate new PNGs with the right size from the SVG with `ksvgtopng5`.

**Comment by hfiguiere** [comment]
a SVG installed in the right place is enough, the export process should do it.

BTW it's bug if the source SVG isn't in the upstream repository

**Comment by Amit0617** [comment]
So i need to confirm whether i am understanding it right or not.
I need to push a `hi-apps-krename.svgz` (of 128 pixels) file at `krename/src/pics` in repository
then i need to edit `krename/src/pics/CMakeLists.txt` and update following 
```
ecm_install_icons(ICONS
    32-apps-krename.png
    48-apps-krename.png
    DESTINATION ${KDE_INSTALL_ICONDIR}
    THEME hicolor
)
```
with 
```
ecm_install_icons(ICONS
    32-apps-krename.png
    48-apps-krename.png
    64-apps-krename.png
    128-apps-krename.png
    DESTINATION ${KDE_INSTALL_ICONDIR}
    THEME hicolor
)
```

**Comment by tsdgeos** [comment]
There's no svg for krename as far as i can see, is there?

I don't think it's reasonable to ask you to create one (yes would be nice, but feels severely out of scope here)

**Comment by hfiguiere** [comment]
If there is no such icon, then the solution is as follow:

1. file an issue upstream for them to have a SVG icon, or 128x128
2. skip icon check by adding `"skip-icons-check": true` to a `flathub.json` file. See https://github.com/flathub/flathub/wiki/App-Maintenance#flathubjson for the flathub.json

**Comment by Amit0617** [comment]
bot, build org.kde.krename

File: flathub.json [hunk 8]
```diff
@@ -0,0 +1 @@
+"skip-icons-check":true
```
Comment (hfiguiere) [inline flathub.json:1]: This should be a valid json file.

**Comment by Amit0617** [comment]
bot, build org.kde.krename

**Pull request closed without merging**